projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/dashboard-filter-bar.component.html

```diff
@@ -0,0 +1,114 @@
+<div class="time-container">
+  <div class="ranger-section" [class.compact]="compactView">
+    <step-execution-time-selection [context]="context"></step-execution-time-selection>
+  </div>
+  <div>
+    <step-time-range-picker
+      (selectionChange)="handleTimeRangeChange($event)"
+      [selectOptions]="timeRangeOptions"
+      [activeSelection]="activeTimeRange"
+      [compact]="compactView"
+    >
+    </step-time-range-picker>
+  </div>
+</div>
+
+<div class="filters-section">
+  <step-alert type="warning" class="warning" *ngIf="rawMeasurementsModeActive">
+    You are using custom filters which may slow down the dashboard.
+  </step-alert>
+
+  <div class="settings-section" *ngIf="compactView">
+    <mat-button-toggle-group [(ngModel)]="oqlModeActive" (change)="toggleOQLMode()">
+      <mat-button-toggle [value]="false"> Basic</mat-button-toggle>
+      <mat-button-toggle [value]="true"> OQL</mat-button-toggle>
+    </mat-button-toggle-group>
+    <step-timeseries-grouping
+      [dimensions]="activeGrouping"
+      (groupingChange)="handleGroupingChange($event)"
+    ></step-timeseries-grouping>
+  </div>
+
+  <div class="filters">
+    <mat-menu #addFilterMenu="matMenu">
+      <ng-container *ngFor="let option of filterOptions">
+        <button mat-menu-item *ngIf="!option.isHidden" (click)="addFilterItem(option)">
```

Comment:
1. The optimal way would be to filter the array, instead of check each element's isHidden flag.
The additional pipe can be created to to the filtration
2. Add trackBy

Reply:
@dvladir this list will never have more than 5-10 items. If there is already a pipe for filtering I'd use it, but otherwise I don't think it make sense to create one for this. Track by identity is ok here I guess.

Reply:
trackBy should be added, filtering the array instead of using ngIf is more of a style question, I would prefer that as well.

Reply:
@neogucky filtering using a custom pipe?

Reply:
@lucianbaciu yes, I think it will be used in multiple places so makes definitly sense